### PR TITLE
N8N-4126 credentials injection and testing on specific nodes

### PR DESCRIPTION
**Pull request by agobrech**

### CODE REVIEW

File: packages/nodes-base/credentials/UProcApi.credentials.ts [hunk 1]
```diff
@@ -19,4 +25,18 @@ export class UProcApi implements ICredentialType {
...
+		credentials: ICredentialDataDecryptedObject,
+		requestOptions: IHttpRequestOptions,
+	): Promise<IHttpRequestOptions> {
+		const token = Buffer.from(`${credentials.email}:${credentials.apiKey}`).toString('base64');
+		requestOptions.headers!['Authorization'] = `Basic ${token}`;
+		return requestOptions;
+	}
+	test: ICredentialTestRequest = {
+		request: {
+			baseURL: 'https://api.uproc.io/api/v2/process',
```
Comment (krynble) [inline packages/nodes-base/credentials/UProcApi.credentials.ts:38]: uProc credential test does not seem to work since /me does not exist (returns 404)

**Review by krynble — CHANGES_REQUESTED** [review]
Hey Ael!

Thanks for all the effort. I've found some things not working fine and I've left a few comments.

File: packages/nodes-base/credentials/LemlistApi.credentials.ts [hunk 2]
```diff
@@ -12,4 +18,19 @@ export class LemlistApi implements ICredentialType {
 			default: '',
 		},
 	];
+	async authenticate(
+		credentials: ICredentialDataDecryptedObject,
+		requestOptions: IHttpRequestOptions,
+	): Promise<IHttpRequestOptions> {
+		const encodedApiKey = Buffer.from(':' + credentials.apiKey).toString('base64');
+		requestOptions.headers!['Authorization'] = `Basic ${encodedApiKey}`;
+		requestOptions.headers!['user-agent'] = 'n8n';
```
[the remark this answers is not in the document]
Comment (agobrech) [reply packages/nodes-base/credentials/LemlistApi.credentials.ts:27]: It was in the old authenticate method so I kept it just in case.

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.192.0`

**Pull request merged**